src/client/common/process/baseDaemon.ts

```diff
@@ -177,14 +177,20 @@ export abstract class BasePythonDaemon {
         return Object.keys(options).every((item) => daemonSupportedSpawnOptions.indexOf(item as any) >= 0);
     }
     protected sendRequestWithoutArgs<R, E, RO>(type: RequestType0<R, E, RO>): Thenable<R> {
-        return Promise.race([this.connection.sendRequest(type), this.connectionClosedDeferred.promise]);
+        if (this.proc && this.proc.exitCode === null) {
```

Comment:
```suggestion
        if (this.proc && typeof this.proc.exitCode !== 'number') {
```

Else if we have `undefined`, then above condition doesn't work. I feel that's better than hardcoding `null`.
**Optional change requets**

Reply:
The api says it returns a number or null. Never undefined.

Reply:
Yes, just that usage of `null` is frowned upon & comparing against `null` isn't a good practice either. 
I guess its old code that they can't remove..